src/geometry.cpp

```diff
@@ -91,11 +91,25 @@ find_cell_inner(Particle& p, const NeighborList* neighbor_list)
         break;
       }
     }
+
+    // If we're attempting a neighbor list search and fail, we
+    // now know we should return false. This will trigger an
+    // exhaustive search from neighbor_list_find_cell and make
+    // the result from that be appended to the neighbor list.
+    if (!found)
+      return false;
```

Comment:
A few lines down (at line 140) we have a similar `!found` check. That one returns `found`, this one returns `false`. Might make things slightly easier to read if both logic checks were identical.

Reply:
True. I think it's more expressive to put `return found` so, I've changed that.